lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2103,6 +2103,89 @@ void AttributeChecker::visitNonOverrideAttr(NonOverrideAttr *attr) {
 }
 
 // SWIFT_ENABLE_TENSORFLOW
+static FuncDecl *resolveAutoDiffAssociatedFunction(
+    TypeChecker &TC, DifferentiableAttr::DeclNameWithLoc specifier,
+    FuncDecl *original, bool isPrimal, Type expectedTy,
+    std::function<bool(FuncDecl *)> isValid) {
+  auto nameLoc = specifier.Loc.getBaseNameLoc();
+  auto overloadDiagnostic = [&]() {
+    if (isPrimal) {
+      TC.diagnose(nameLoc, diag::differentiable_attr_primal_overload_not_found,
+                  specifier.Name, expectedTy);
+    } else {
+      TC.diagnose(nameLoc, diag::differentiable_attr_overload_not_found,
+                  specifier.Name, expectedTy);
+    }
+  };
+  auto ambiguousDiagnostic = [&]() {
+    TC.diagnose(nameLoc,
+                diag::differentiable_attr_ambiguous_function_identifier,
+                specifier.Name);
+  };
+  auto notFunctionDiagnostic = [&]() {
+    TC.diagnose(nameLoc, diag::differentiable_attr_specified_not_function,
+                specifier.Name);
+  };
+  std::function<void()> invalidTypeContextDiagnostic = [&]() {
```

Comment:
```suggestion
  auto invalidTypeContextDiagnostic = [&]() {
```

Reply:
This seems to cause an error:
```
/Users/swiftci/jenkins/workspace/swift-PR-TensorFlow-macOS/swift/lib/Sema/TypeCheckAttr.cpp:2181:43: error: no viable conversion from '(lambda at /Users/swiftci/jenkins/workspace/swift-PR-TensorFlow-macOS/swift/lib/Sema/TypeCheckAttr.cpp:2129:39)' to 'const Optional<std::function<void ()> >'
      lookupOptions, hasValidTypeContext, invalidTypeContextDiagnostic);
                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I'll change it back.

Reply:
Yep